ArjunaCore/arjuna/classes/com/arjuna/ats/arjuna/common/CoordinatorEnvironmentBean.java

```diff
@@ -519,6 +521,47 @@ public void setTxReaperZombieMax(int txReaperZombieMax)
         this.txReaperZombieMax = txReaperZombieMax;
     }
 
+    /**
+     * Returns the number of milliseconds delay after a transaction is started,
+     * before the reaper will start taking periodic stack traces from it.
+     *
+     * Default: 180000 (3 minutes)
+     *
+     * @return the reaper tracing grace period, in milliseconds.
+     */
+    public long getTxReaperTraceGracePeriod() {
```

Comment:
A remark, the grace period will avoid generating StackTraceCapture object instances (which could be thousands for high throughput) when the transactions are short running. I wonder if there are other reasons for the grace period.

Reply:
Nope, that is indeed pretty much what it's intended to do. Credit to Stephen Fikes for that idea.